src/main/java/mate/academy/hibernate/relations/dao/impl/CountryDaoImpl.java

```diff
@@ -12,11 +15,34 @@ public CountryDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Country add(Country country) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(country);
+            transaction.commit();
+        } catch (RuntimeException e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException(String.format(
+                    "Cannot save country: %s to DB", country.getName()), e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
```

Comment:
The checklist recommends using 'try with resources' for read operations and 'try with finally' for transactions. Consider refactoring this to use 'try with resources' to ensure that the session is always closed, even if an exception occurs.